VmAgent.Core/Interfaces/SystemOperations.cs

```diff
@@ -261,5 +300,50 @@ public void SetUnixFilePermissions(string filePath, int permissions)
             var unixFileInfo = new UnixFileInfo(filePath);
             unixFileInfo.FileAccessPermissions = (FileAccessPermissions)permissions;
         }
+
+        public void SetUnixOwnerIfNeeded(string path, bool applyToAllContents = false)
+        {
+            if (_vmConfiguration != null && _vmConfiguration.RunContainersInUsermode)
+            {
+                _logger.LogInformation($"Setting unix owner for {path}");
+
+                _fileSystemOperations.SetUnixOwner(path, User);
+
+                if (_fileSystemOperations.IsDirectory(path) && applyToAllContents)
+                {
+                    DirectoryInfo dirInfo = new DirectoryInfo(path);
+                    IEnumerable<FileSystemInfo> childItems = _fileSystemOperations.GetFileSystemInfos(dirInfo, true);
+
+                    foreach (FileSystemInfo child in childItems)
+                    {
+                        _fileSystemOperations.SetUnixOwner(child.FullName, User);
+                    }
+                }
+            }
+            else
+            {
+                _logger.LogInformation("Unix file setting not needed");
+            }
+        }
+
+        public void CreateDirectoryAndParents(DirectoryInfo directory)
+        {
+            //It's possible for the caller to unknowingly call with Director.Parent which can be null
+            if(directory == null)
```

Comment:
This seems weird. If it is in our code, we should pass in the appropriate one. The fact that we silently return here is a code smell

Reply:
Artifact from before I wrote the try method, the null was also the recursion escape, but it's not needed now